gradio/chat_interface.py

```diff
@@ -82,8 +84,9 @@ def __init__(
         """
         Parameters:
             fn: The function to wrap the chat interface around. Should accept two parameters: a string input message and list of two-element lists of the form [[user_message, bot_message], ...] representing the chat history, and return a string response. See the Chatbot documentation for more information on the chat history format.
+            multimodal: If True, the chat interface will use a gr.MultimodalTextbox component for the input, which allows for the uploading of multimedia files. If False, the chat interface will use a gr.Textbox component for the input.
```

Comment:
I think the `Undo` and `Retry` do not work if `multimodal=True`. We're passing in a string instead of a dict. This is the error I got for Retry

```
  File "/Users/freddy/miniconda3/envs/gradio-310/lib/python3.10/site-packages/anyio/_backends/_asyncio.py", line 807, in run
    result = context.run(func, *args)
  File "/Users/freddy/sources/gradio/gradio/utils.py", line 493, in run_sync_iterator_async
    return next(iterator)
  File "/Users/freddy/sources/scratch/multimodal_textbox_demo.py", line 5, in test
    t = x['text']
TypeError: string indices must be integers
```